STICKY: GCC 6

alarcher opened the pull request:
Tracking progress on GCC 7-built userland.

alarcher committed:
Use GCC 6

pyhalov commented on components/network/isc-dhcp/Makefile:
```diff
@@ -23,28 +23,32 @@
 COMPONENT_NAME=		isc-dhcp
-COMPONENT_VERSION=	4.3.4
-IPS_COMPONENT_VERSION=	4.3.4
+COMPONENT_VERSION=	4.3.5
```
Let's merge this update. (I hope, it doesn't break 4.9 build)

pyhalov commented on components/library/dbus/Makefile:
```diff
@@ -16,29 +16,31 @@
 
 COMPONENT_NAME= dbus
-COMPONENT_VERSION= 1.10.12
+COMPONENT_VERSION= 1.10.18
```
Should we merge this (dbus update)? There's already 1.10.20

alarcher commented:
@pyhalov @AndWac @xen0l More fixes to cherry-pick...

alarcher committed:
luajit: bump to 2.0.5

alarcher committed:
jruby: bump to 9.1.15.0

alarcher committed:
thunderbird: fix runpath resolution

pyhalov committed:
qt4: fix GCC-6 build

alarcher committed:
clang: fix with gcc-6

alarcher committed:
qt5: fix gcc-6 build

alarcher committed:
gmp: fix runpath

pyhalov committed:
DBI-PostgreSQL: fix GCC 6 build

pyhalov committed:
DBI-MySQL: fix GCC 6 build

alarcher committed:
Bump RELEASE_MAJOR

alarcher committed:
tigervnc: fix gcc-6 build

alarcher committed:
gcc3: disable parallel build

richlowe committed:
make-rules: use 'pkgdepend resolve -e' and REQUIRED_PACKAGES

The pkg(5) in OI is new enough now to allow the use of 'pkgdepend
resolve -e' to vastly reduce build times at the expense of manually
maintaining a list of dependencies.

This change allows that to happen in the "new" way, using the
REQUIRED_PACKAGES list that most Makefiles already contain, rather than
the "old" way way, using a static resolve.deps which was never actively
used in OI.

pyhalov committed:
make-rules: add component's fmris to RESOLVE_DEPS

alarcher committed:
Fix desktop/xchm REQUIRED_PACKAGES

alarcher committed:
Fix developer/cppunit REQUIRED_PACKAGES

alarcher committed:
Fix developer/opensp REQUIRED_PACKAGES

alarcher committed:
Fix libmusicbrainz-5 REQUIRED_PACKAGES

alarcher committed:
Fix libofa REQUIRED_PACKAGES

alarcher committed:
Fix botan REQUIRED_PACKAGES

alarcher committed:
Fix cxxtools REQUIRED_PACKAGES

alarcher committed:
Fix id3lib REQUIRED_PACKAGES

alarcher committed:
Fix lcms REQUIRED_PACKAGES

alarcher committed:
Fix libconfig REQUIRED_PACKAGES

alarcher committed:
Fix libsodium REQUIRED_PACKAGES

alarcher committed:
Fix mozilla-nspr REQUIRED_PACKAGES

alarcher committed:
Fix rarian REQUIRED_PACKAGES

alarcher committed:
Fix stunnel REQUIRED_PACKAGES

alarcher committed:
Fix dvd+rw-tools REQUIRED_PACKAGES

alarcher committed:
Fix gptfdisk REQUIRED_PACKAGES

alarcher committed:
Fix testdisk REQUIRED_PACKAGES

alarcher committed:
Fix groff REQUIRED_PACKAGES

alarcher committed:
Convert gcc-4-runtime to gcc-6-runtime

alarcher committed:
Convert g++-4-runtime to g++-6-runtime

alarcher committed:
Convert gfortran-4-runtime to gfortran-6-runtime

pyhalov commented on components/developer/luajit/patches/luajit-01-Makefile.patch:
```diff
@@ -14,3 +14,13 @@
  INSTALL_LIB=   $(DPREFIX)/$(MULTILIB)
  INSTALL_SHARE= $(DPREFIX)/share
  INSTALL_INC=   $(DPREFIX)/include/luajit-$(MAJVER).$(MINVER)
+318c318,324
```
Please, fix it to be in unified format

pyhalov commented on components/fix-stuff.mk:
```diff
@@ -0,0 +1,30 @@
+USERLAND_COMPONENTS_DIR ?= .
```
Please, don't add this file to oi-userland

pyhalov commented on components/openindiana/illumos-gate/Makefile:
```diff
@@ -19,18 +19,18 @@ COMPONENT_NAME=		illumos-gate
 COMPONENT_SRC=		$(COMPONENT_NAME)
```
Avoid adding this change to oi-userland

pyhalov commented on make-rules/ips-buildinfo.mk:
```diff
@@ -36,7 +36,7 @@ RELEASE_MAJOR ?= 2017
 # Release minor number: 0, 1, 2, etc.
 #
-RELEASE_MINOR ?= 0
+RELEASE_MINOR ?= 1
```
@@Don't bump RELEASE_MINOR, better update RELEASE_MAJOR

pyhalov commented on components/Makefile:
```diff
@@ -155,3 +155,31 @@ $(COMPONENT_DIRS.nosetup):	$(WS_LOGS) FORCE
 
 FORCE:
+
+list-unpublished:
```
Do we need it in oi-userland?

alarcher commented:
@pyhalov I'll remove the commits that are not needed when we decide to merge. Until then since I build on two machines I need to have a branch with the stuff I need.

alarcher committed:
Fix cluster/cluster-glue REQUIRED_PACKAGES

alarcher committed:
Fix caja REQUIRED_PACKAGES

alarcher committed:
Fix cluster/resource-agents REQUIRED_PACKAGES

alarcher committed:
Fix gcc-7 REQUIRED_PACKAGES

alarcher committed:
Fix percona-server-57 REQUIRED_PACKAGES

alarcher committed:
Fix subversion REQUIRED_PACKAGES

alarcher committed:
Fix geeqie REQUIRED_PACKAGES

alarcher committed:
Fix plplot REQUIRED_PACKAGES

alarcher committed:
Fix imagemagick REQUIRED_PACKAGES

alarcher committed:
Fix fbreader with gcc-6 and REQUIRED_PACKAGES

alarcher committed:
Fix graphviz REQUIRED_PACKAGES

alarcher committed:
Fix glibmm REQUIRED_PACKAGES

alarcher committed:
Fix libmusicbrainz-3 REQUIRED_PACKAGES

alarcher committed:
Fix protobuf REQUIRED_PACKAGES

alarcher committed:
Fix mpich REQUIRED_PACKAGES

alarcher committed:
Fix asterisk REQUIRED_PACKAGES

alarcher committed:
Fix lftp REQUIRED_PACKAGES

alarcher committed:
Fix samba REQUIRED_PACKAGES

alarcher committed:
Fix cups REQUIRED_PACKAGES

alarcher committed:
Fix cups-filters REQUIRED_PACKAGES and build with libjpeg8

alarcher committed:
boost: disable fiber with gcc-6

alarcher committed:
Fix php-7_0 REQUIRED_PACKAGES

alarcher committed:
openjdk-8: update to 1.8.152 and fix gcc-6 build

alarcher committed:
Fix mariadb-101 REQUIRED_PACKAGES

alarcher committed:
Fix couchdb-21 REQUIRED_PACKAGES

alarcher committed:
librecad: fix gcc-6 build

pyhalov commented on components/database/mongodb-33/Makefile:
```diff
@@ -77,12 +77,16 @@ install:	$(INSTALL_64) $(INSTALL_TOOLS)
-REQUIRED_PACKAGES += SUNWcs
+# Build dependencies
 REQUIRED_PACKAGES += developer/build/scons
-REQUIRED_PACKAGES += developer/gcc-7
```
Please, preserve developer/gcc-7 build dependency

pyhalov commented on components/database/mongodb-34/Makefile:
```diff
@@ -107,13 +107,17 @@ install:	$(INSTALL_64) $(INSTALL_TOOLS)
+# Build dependencies
 REQUIRED_PACKAGES += developer/build/scons
-REQUIRED_PACKAGES += developer/gcc-7
 REQUIRED_PACKAGES += developer/golang
```
please, preserve developer/gcc-7 build dependency

alarcher committed:
mongodb-33: use gcc-6

alarcher committed:
mongodb-34: use gcc-6

alarcher committed:
Fix zenity REQUIRED_PACKAGES

pyhalov committed:
nvidia: use GCC 6 runtime libraries

alarcher committed:
Fix flac REQUIRED_PACKAGES

alarcher committed:
Fix gst-plugins-good1 REQUIRED_PACKAGES

alarcher committed:
Fix openvpn REQUIRED_PACKAGES

alarcher committed:
Fix pgbadger REQUIRED_PACKAGES

alarcher committed:
Fix ccsm REQUIRED_PACKAGES

alarcher committed:
Fix python27 REQUIRED_PACKAGES

alarcher committed:
Fix erlang REQUIRED_PACKAGES

alarcher committed:
Fix arpack REQUIRED_PACKAGES

alarcher committed:
Fix openblas REQUIRED_PACKAGES

alarcher committed:
Fix mesa REQUIRED_PACKAGES

alarcher committed:
Fix ipmitool REQUIRED_PACKAGES

alarcher committed:
Fix gnupg REQUIRED_PACKAGES

alarcher committed:
Fix diveintopython REQUIRED_PACKAGES

alarcher committed:
xcowsay: fix build with automake 1.15.1

alarcher committed:
Fix xf86-video-intel REQUIRED_PACKAGES

alarcher committed:
Fix ddu REQUIRED_PACKAGES

alarcher committed:
Fix sbcl REQUIRED_PACKAGES

alarcher committed:
Fix gettext REQUIRED_PACKAGES

alarcher committed:
Fix itstool REQUIRED_PACKAGES

alarcher committed:
Fix curl REQUIRED_PACKAGES

alarcher committed:
Fix apache2-modules/mod_security2 REQUIRED_PACKAGES

alarcher committed:
Fix pinentry REQUIRED_PACKAGES

alarcher committed:
haproxy: fix gcc-6 build

alarcher committed:
Fix gst-plugins-bad REQUIRED_PACKAGES

alarcher committed:
gcc49: build with self

alarcher committed:
ruby-22: use gcc-6

alarcher committed:
ruby-23: use gcc-6

pyhalov commented:
I've checked that after Ruby fixes augeas ugly hack is not necessary

alarcher committed:
illumos-gcc: build with gcc49 and do not use system gmp and mpfr libs

pyhalov committed:
libtheora: fix crash in oc_state_init

pyhalov committed:
Fix gst-plugin-base REQUIRED_PACKAGES

alarcher committed:
expect: fix with gcc-6

alarcher committed:
openjdk-7: fix with gcc-6

alarcher committed:
gcc-5: fix build with gcc-6

alarcher committed:
firefox: fix runpath with gcc-6

pyhalov committed:
tmux: fix tty handling with gcc-6

alarcher committed:
Fix service-hacluster REQUIRED_PACKAGES

alarcher committed:
Fix slim_source REQUIRED_PACKAGES

alarcher committed:
tomcat-8: pass UTF-8 encoding options to javac

alarcher committed:
Fix tomcat REQUIRED_PACKAGES
